lib/Raven/HttpClient/HttpClientFactoryInterface.php

```diff
@@ -0,0 +1,24 @@
+<?php
+
+namespace Raven\HttpClient;
+
+use Http\Client\HttpAsyncClient;
+
+/**
+ * This interface implements a contract that should be respected by all factories
+ * willing to create HTTP client instances.
+ *
+ * @author Stefano Arlandini <sarlandini@alice.it>
+ */
+interface HttpClientFactoryInterface
```

Comment:
I do not think you need to create a HTTP client. You should require a configured client or fallback to client discovery.